libs/unity/library/Runtime/Shaders/YUVFeedShaderUnlit.shader

```diff
@@ -1,90 +1,82 @@
-                UNITY_INITIALIZE_OUTPUT(v2f, o);
-                UNITY_INITIALIZE_VERTEX_OUTPUT_STEREO(o);
-                o.vertex = UnityObjectToClipPos(v.vertex);
-                o.uv = v.uv;
+			struct appdata
+			{
+				float4 vertex : POSITION;
+				float2 uv : TEXCOORD0;
+			};
 
-                // Flip texture coordinates vertically.
-                // Texture2D.LoadRawTextureData() always expects a bottom-up image, but the MediaPlayer
-                // upload code always get a top-down frame from WebRTC. The most efficient is to upload
-                // as is (inverted) and revert here.
-                o.uv.y = 1 - v.uv.y;
+			struct v2f
+			{
+				float2 uv : TEXCOORD0;
+				float4 vertex : SV_POSITION;
+			};
 
+			v2f vert (appdata v)
+			{
+				v2f o;
+				o.vertex = UnityObjectToClipPos(v.vertex);
+				o.uv = v.uv;
+#if !UNITY_UV_STARTS_AT_TOP
+				o.uv.y = 1 - v.uv.y;
+#endif
 #ifdef MIRROR
-                // Optional left-right mirroring (horizontal flipping)
-                o.uv.x = 1 - v.uv.x;
+				o.uv.x = 1 - v.uv.x;
+#endif
+#ifdef VMIRROR
+				o.uv.y = 1 - v.uv.y;
 #endif
-                return o;
-            }
+				return o;
+			}
 
-            sampler2D _YPlane;
-            sampler2D _UPlane;
-            sampler2D _VPlane;
+			sampler2D _YPlane;
+			sampler2D _UPlane;
+			sampler2D _VPlane;
 
-            half3 yuv2rgb(half3 yuv)
-            {
-                // The YUV to RBA conversion, please refer to: http://en.wikipedia.org/wiki/YUV
-                // Y'UV420p (I420) to RGB888 conversion section.
-                half y_value = yuv[0];
-                half u_value = yuv[1];
-                half v_value = yuv[2];
-                half r = y_value + 1.370705 * (v_value - 0.5);
-                half g = y_value - 0.698001 * (v_value - 0.5) - (0.337633 * (u_value - 0.5));
-                half b = y_value + 1.732446 * (u_value - 0.5);
-                return half3(r, g, b);
-            }
+			float3 yuv2rgb(float3 yuv)
+			{
+				// The YUV to RBA conversion, please refer to: http://en.wikipedia.org/wiki/YUV
+				// Y'UV420p (I420) to RGB888 conversion section.
+				float y_value = yuv[0];
+				float u_value = yuv[1];
+				float v_value = yuv[2];
+				float r = y_value + 1.370705 * (v_value - 0.5);
+				float g = y_value - 0.698001 * (v_value - 0.5) - (0.337633 * (u_value - 0.5));
+				float b = y_value + 1.732446 * (u_value - 0.5);
+				return float3(r, g, b);
+			}
 
-            fixed3 frag (v2f i) : SV_Target
-            {
-                half3 yuv;
-                yuv.x = tex2D(_YPlane, i.uv).r;
-                yuv.y = tex2D(_UPlane, i.uv).r;
-                yuv.z = tex2D(_VPlane, i.uv).r;
-                return yuv2rgb(yuv);
-            }
-            ENDCG
-        }
-    }
+			fixed4 frag (v2f i) : SV_Target
+			{
+				float3 yuv;
+				yuv.x = tex2D(_YPlane, i.uv).r;
+				yuv.y = tex2D(_UPlane, i.uv).r;
+				yuv.z = tex2D(_VPlane, i.uv).r;
+				return fixed4(yuv2rgb(yuv), 1.0);
+			}
```

Comment:
For curiosity, why this change (half -> float, fixed3 -> fixed4)? @djee-ms any reason why we were using half/fixed3?

Reply:
I changed it to half. I may have typed float out of habit. Its rare this makes a difference on modern hardware but probably better to use lowest precision anyways.